File: Model/triangolo.cpp
#include "triangolo.h"
#include <QVector>
#include <math.h>

Triangolo::Triangolo(const Punto& p1, const Punto& p2, const Punto& p3) : LatiFiniti() {
    pushVertice(p1);
    pushVertice(p2);
    pushVertice(p3);
}

double Triangolo::getArea() const {
    double semiP = getPerimetro()/2;
    QVector<double> l = getLati();
    double area = sqrt(semiP * (semiP-l[0]) * (semiP-l[1]) * (semiP-l[2]));
    return area;
}

double Triangolo::getAltezza() const {
    return (getArea()/2)/getLato(1);
}

Triangolo* Triangolo::clone() const {
    return new Triangolo(*this);
}

File: SuperQtKalkJAVA/src/Triangolo.java
import java.util.ArrayList;
import static java.lang.Math.sqrt;

public class Triangolo extends LatiFiniti {

    public Triangolo(Punto a, Punto b, Punto c) {
        super(new ArrayList<Punto>());
        pushVertice(a);
        pushVertice(b);
        pushVertice(c);
    }

    public double getArea() {
        double semiP = getPerimetro()/2;
        ArrayList<Double> l = getLati();
        double area = sqrt(semiP * (semiP-l.get(0)) * (semiP-l.get(1)) * (semiP-l.get(2)));
        return area;
    }
    

    public double getAltezza() {
        return getArea()/getLato(1);
    }
}
